src/dune_pkg/fetch.ml

```diff
@@ -2,7 +2,12 @@ open Import
 open Fiber.O
 
 module Curl = struct
-  let bin = lazy (Bin.which ~path:(Env_path.path Env.initial) "curl")
+  let bin =
+    lazy
+      (match Bin.which ~path:(Env_path.path Env.initial) "curl" with
+       | Some p -> p
+       | None -> User_error.raise [ Pp.text "curl not available in PATH" ])
```

Comment:
If you want to, you can use the new function to denote `curl` is a command; you can see that in #9079. Also maybe add a hint to install `curl`?

Reply:
AFAIK every platform we support should have curl. Do you have a platform in mind where this is not the case?

Reply:
If a user is using a system that doesn't come preinstalled with curl, then I think there is a good chance they know what they are doing and we can't really give a helpful hint. I'm open to improving this if you have a suggestion for a hint, but I cannot think of one here.